Sources/Apexy/Endpoint.swift

```diff
@@ -29,6 +32,16 @@ public protocol Endpoint {
     /// - Returns: A new endpoint content.
     /// - Throws: Any error creating content.
     func content(from response: URLResponse?, with body: Data) throws -> Content
+    
+    /// Obtain error from response with body.
+    ///
+    /// - Parameters:
+    ///   - response: The metadata associated with the response.
+    ///   - body: The response body.
+    ///   - error: The response error.
+    /// - Returns: A new endpoint error.
+    /// - Throws: Any error creating error.
+    func error(fromResponse response: URLResponse?, withBody body: Data?, withError error: Error) -> ErrorType
```

Comment:
This method might have a default implementation.

```swift
extension Endpoint where ErrorType == Error {
    func error(fromResponse response: URLResponse?, withBody body: Data?, withError error: Error) -> Error {
         return error
     }
}
```

Reply:
it is strange that the `makeRequest()` and `content(from:with:)` do not have a typed error.

it would be possible to combine `content(from:with:)` with `error(fromResponse:withBody: withError:)`

```swift
func decode(response: URLResponse?, body: Data?, error: Error?) -> Result<Content, Error>
```

but I do not know how it will be better with method `makeRequest()`

Reply:
`throw` doesn't help at all to specify an error for specific endpoint. So i'll try to remove all `throw` from `Endpoint` methods and use `Result` instead

Reply:
But it will lead to significant changes🤬

Reply:
maybe leave only the response processing error typed ?

Reply:
I changed `makeRequest` methods and merged `content` and `error` methods. It look ok i think. But i caused huge api changing